libr/magic/apprentice.c

```diff
@@ -1801,7 +1801,10 @@ static int apprentice_map(RMagic *ms, struct r_magic **magicp, ut32 *nmagicp, co
 	}
 
 #ifdef QUICK
-	if ((mm = mmap (0, (size_t)st.st_size, PROT_READ, //OPENBSDBUG  |PROT_WRITE,
+#ifndef PROT_MAX
+#define PROT_MAX(p) 0
+#endif
+	if ((mm = mmap (0, (size_t)st.st_size, PROT_READ|PROT_MAX(PROT_READ), //OPENBSDBUG  |PROT_WRITE,
```

Comment:
i dont see why we should be calling PROT_MAX here, with PROT_READ twice aand an OR seems like a nop to me. Imho we can safely remove all this |PROT_MAX(PROT_READ*....

Reply:
hmmm PROT_MAX is definitely not a no-op but I could simplify a bit this bit tho.

Reply:
whats the benefit of introducing this PROT_MAX in a compile-time defined option for mmap? imho prot-max is useful when its controlled by the user. or maybe  im missing  s omething

Reply:
you re not wrong, even though it s masinly because the address is available thus I did not apply this everywhere in the base code but that s fine if this does not get merged tho, just proposal ;-)